app/services/reaction_handler.rb

```diff
@@ -15,46 +15,51 @@ def success?
     end
   end
 
+  def self.create(params, current_user:)
+    new(params, current_user: current_user).create
+  end
+
   def self.toggle(params, current_user:)
     new(params, current_user: current_user).toggle
   end
 
   def initialize(params, current_user:)
     @params = params
+    @reactable_id = params[:reactable_id]
+    @reactable_type = params[:reactable_type]
     @current_user = current_user
     @category = params[:category] || "like"
   end
 
-  attr_reader :category, :params, :current_user
+  attr_reader :category, :params, :current_user, :reactable_type, :reactable_id
 
   delegate :rate_limiter, to: :current_user
 
+  def create
+    destroy_contradictory_mod_reactions if reactable_type == "Article"
+    return noop_result if existing_reaction
+
+    create_new_reaction
+  end
+
   def toggle
-    if params[:reactable_type] == "Article" && params[:category].in?(Reaction::PRIVILEGED_CATEGORIES)
-      destroy_contradictory_mod_reactions(
-        params[:reactable_id],
-        params[:reactable_type],
-        current_user,
-        params[:category],
-      )
-    end
+    destroy_contradictory_mod_reactions if reactable_type == "Article"
+    return handle_existing_reaction if existing_reaction
 
-    existing_reaction = get_existing_reaction
-    if existing_reaction
-      handle_existing_reaction(existing_reaction)
-    else
-      create_new_reaction
-    end
+    create_new_reaction
   end
 
   private
 
-  def destroy_contradictory_mod_reactions(id, type, mod, category)
+  def destroy_contradictory_mod_reactions
+    return unless params[:category].in?(Reaction::PRIVILEGED_CATEGORIES)
```

Comment:
Should this just be `category`

```suggestion
    return unless category.in?(Reaction::PRIVILEGED_CATEGORIES)
```